pkg/controllers/management/auth/globalroles/globalrole_handler.go

```diff
@@ -196,6 +217,134 @@ func (gr *globalRoleLifecycle) reconcileCatalogRole(globalRole *v3.GlobalRole) e
 	return nil
 }
 
+// reconcileNamespacedRoles ensures that Roles exist in each namespace of NamespacedRules
+func (gr *globalRoleLifecycle) reconcileNamespacedRoles(globalRole *v3.GlobalRole) error {
+	var returnError error
+	globalRoleName := wrangler.SafeConcatName(globalRole.Name)
+
+	// For collecting all the roles that should exist for the GlobalRole
+	roleUIDs := map[types.UID]struct{}{}
+
+	for ns, rules := range globalRole.NamespacedRules {
+		namespace, err := gr.nsLister.Get("", ns)
+		if apierrors.IsNotFound(err) {
```

Comment:
Might be good to check if the returned namespace was nil
```suggestion
		if apierrors.IsNotFound(err) || namespace == nil{
```